src/hotspot/share/opto/subnode.cpp

```diff
@@ -1529,10 +1529,46 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
     }
   }
 
+  const int cmp1_op = cmp1->Opcode();
+  const int cmp2_op = cmp2->Opcode();
```

Comment:
Use these in previous code too by move them to line `#1485` and replacing `op2`.